### PR TITLE
build(azure): Fix type generation for ESM builds

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR updates the shared tsc-multi settings and also updates the azure packages to produce correct types for their ESM build. I also removed the attw overrides, so now that attw is passing it means that these changes addressed the "ESM masquerading as CJS" problems the packages currently have.

### CODE REVIEW

File: azure/packages/azure-client/package.json [hunk 1]
```diff
@@ -30,12 +30,15 @@
 		"api": "fluid-build . --task api",
 		"api-extractor:commonjs": "api-extractor run --local",
 		"api-extractor:esnext": "copyfiles -u 1 \"dist/**/*-@(alpha|beta|public|untrimmed).d.ts\" lib",
+		"api-extractor:rename-types": "renamer \"lib/**\" -f .d.ts -r .d.mts --force",
+		"api-extractor:rewrite-type-imports": "replace-in-file \"/from \\\"\\\\.((.*?)(\\\\.mjs)?)\\\";/g\" \"from \\\".$2.mjs\\\";\" \"lib/**/*.d.mts\" --isRegex",
```
Comment (tylerbutler) [inline azure/packages/azure-client/package.json:34]: I wish this didn't take so much escaping, but it's easier than writing and shipping a tool to do it.

File: fluidBuild.config.cjs [hunk 2]
```diff
@@ -55,13 +55,25 @@ module.exports = {
...
 		},
 		"api-extractor:commonjs": [...tscDependsOn, "tsc"],
 		"api-extractor:esnext": [...tscDependsOn, "api-extractor:commonjs", "build:esnext"],
-		"build:docs": [...tscDependsOn, "tsc"],
-		"ci:build:docs": [...tscDependsOn, "tsc"],
+		"build:rename-types": ["api-extractor:esnext"],
+		"build:rewrite-type-imports": ["build:rename-types"],
+		"build:docs": [...tscDependsOn, "tsc", "build:rename-types", "build:rewrite-type-imports"],
+		"ci:build:docs": [
+			...tscDependsOn,
```
Comment (Josmithr) [inline fluidBuild.config.cjs:72]: Do these tasks that depend directly on `tsc` also need to depend on `tscDependsOn`?

**Comment by Josmithr** [comment]
> These tasks are wired up to run as part of the docs build, but I'm not sure that makes sense. Suggestions welcome.

Seems like they should just be wired into `build` / `tsc`, rather than `build:docs` IMO. But since this is intended as a short term hack, it probably doesn't matter too much.

**Comment by Josmithr** [comment]
Just curious, how easy would this be to implement in `tsc-multi`? Could we contribute this functionality there? I don't want to go too crazy with pnpm patches, but I'm wondering if it would require less infra overhaul to accomplish this that way until tsc-multi adds functionality for this.

File: fluidBuild.config.cjs (hunk 2 above)
Comment (tylerbutler) [reply fluidBuild.config.cjs:72]: No; I opened https://github.com/microsoft/FluidFramework/pull/18686 to address that.

File: .npmrc [hunk 3]
```diff
@@ -10,6 +10,7 @@ workspace-concurrency=0
 
 # Hoist these dependencies to the root workspace
 public-hoist-pattern[]=@arethetypeswrong/cli
+public-hoist-pattern[]=renamer
```
Comment (tylerbutler) [inline .npmrc:13]:
```suggestion
```

File: packages/dds/sequence/tsc-multi.test.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+	"targets": [
+		{
+			"extname": ".cjs",
+			"rewriteDtsImports": true,
```
Comment (tylerbutler) [inline packages/dds/sequence/tsc-multi.test.json:5]:
```suggestion
```

File: common/build/build-common/tsc-multi.esm.json [hunk 5]
```diff
@@ -1,6 +1,12 @@
 {
 	"targets": [
-		{ "extname": ".mjs", "module": "ESNext", "moduleResolution": "node10", "outDir": "./lib" }
+		{
+			"extname": ".mjs",
+			"rewriteDtsImports": true,
+			"module": "ESNext",
```
Comment (tylerbutler) [inline common/build/build-common/tsc-multi.esm.json:6]:
```suggestion
			"module": "Node16",
```

**Comment by tylerbutler** [comment]
Waiting until after 8.0 is released to merge this, so marking as draft

File: azure/packages/test/end-to-end-tests/.eslintrc.cjs [hunk 6]
```diff
@@ -7,6 +7,7 @@ module.exports = {
 	rules: {
 		"prefer-arrow-callback": "off",
 		"@typescript-eslint/strict-boolean-expressions": "off", // requires strictNullChecks=true in tsconfig
+		"import/namespace": "off",
```
Comment (Josmithr) [inline azure/packages/test/end-to-end-tests/.eslintrc.cjs:10]: Can we add a comment explaining why this is disabled?

**Review by Josmithr — APPROVED** [review]

**Pull request merged**